src/components/AttachmentCarousel.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'lodash';
+import Button from './Button';
+import * as Expensicons from './Icon/Expensicons';
+import styles from '../styles/styles';
+import themeColors from '../styles/themes/default';
+import reportActionPropTypes from '../pages/home/report/reportActionPropTypes';
+import CONFIG from '../CONFIG';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+
+const propTypes = {
+    /** sourceUrl is used to determine the starting index in the array of attachments */
+    sourceURL: PropTypes.string,
+
+    /** Object of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+    /** Callback to update the parent modal's state with a sourceUrl and name from the attachments array */
+    onArrowPress: PropTypes.func,
+};
+
+const defaultProps = {
+    sourceURL: '',
+    reportActions: {},
+    onArrowPress: () => {},
+};
+
+class AttachmentCarousel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        let page;
+        const actionsArr = _.values(props.reportActions);
+        const attachments = _.reduce(actionsArr, (attachmentsAccumulator, reportAction) => {
+            if (reportAction.originalMessage && reportAction.originalMessage.html) {
+                const matchesIt = reportAction.originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA);
+                const matches = [...matchesIt];
+                if (matches.length === 2) {
+                    const [src, name] = matches;
+                    if (src[2].includes(props.sourceURL)) {
+                        page = attachmentsAccumulator.length;
+                    }
+                    const url = src[2].replace(
+                        CONFIG.EXPENSIFY.EXPENSIFY_URL,
+                        CONFIG.EXPENSIFY.URL_API_ROOT,
+                    );
+                    attachmentsAccumulator.push({sourceURL: url, file: {name: name[2]}});
+                }
+            }
+            return attachmentsAccumulator;
+        }, []);
+
+        this.state = {
+            page,
+            attachments,
+            isBackDisabled: page === 0,
+            isForwardDisabled: page === attachments.length - 1,
+        };
+
+        this.cycleThroughAttachments = this.cycleThroughAttachments.bind(this);
+    }
+
+    /**
+     * increments or decrements the index to get another selected item
+     * @param {Boolean} shouldDecrement
+    */
+    cycleThroughAttachments(shouldDecrement) {
+        this.setState((prevState) => {
+            const attachments = prevState.attachments;
+            const page = prevState.page;
```

Comment:
could have just destructured `attachments`, `page`

Reply:
Got it, I was going with the idea that I couldn't destructure the state. I could just destructure the argument though?

Reply:
```javascript
this.setState(({attachments, page}) => { ...
```